pandas/io/packers.py

```diff
@@ -485,9 +485,14 @@ def decode(obj):
         def create_block(b):
             values = unconvert(b['values'], dtype_for(b['dtype']),
                                b['compress']).reshape(b['shape'])
+            items = b['items']
+            if isinstance(items, np.ndarray):
```

Comment:
then check if you have ``locs`` first, if you do use it, otherwise use the ``items``. This will provide back-compat w/o having to type infer (e.g. what if we changed ``items`` to an ``np.ndarray``? but that didn't mean location but meant values)